docs/tutorial/alf_snapshot_and_advanced_play.rst

```diff
@@ -0,0 +1,249 @@
+code is located at :mod:`.alf.examples.tutorial.ac_render_conf`.
+
+We will again train a model on the "CartPole-v0" task. So first of all, we import
+all the configuration from :mod:`.alf.examples.ac_cart_pole_conf`:
+
+.. code-block:: python
+
+    from alf.examples import ac_cart_pole_conf
+
+And import the :mod:`~.alf.summary.render` module
+
+.. code-block:: python
+
+    import alf.summary.render as render
+
+Then to tell the :mod:`~.alf.bin.play` module what to render, we overwrite
+``predict_step()`` of the original AC algorithm:
+
+.. code-block:: python
+
+    class ACRenderAlgorithm(ActorCriticAlgorithm):
+       def predict_step(self, inputs, state):
+           alg_step = super().predict_step(inputs, state)
+           action = alg_step.output
+           action_dist = alg_step.info.action_distribution
+           with alf.summary.scope("ACRender"):
+               # Render an action image
+               action_img = render.render_action(
+                   name="predicted_action",
+                   action=action,
+                   action_spec=self._action_spec)
+               # Render an action distribution image
+               action_dist_img = render.render_action_distribution(
+                   name="predicted_action_distribution",
+                   act_dist=action_dist,
+                   action_spec=self._action_spec)
+           # Put the two ``Image`` objects into ``info``. Any nest structure is
+           # acceptable for the new ``info``. ALF's play will look for ``Image``
+           # objects.
+           return alg_step._replace(
+               info=dict(action_img=action_img,
+                         action_dist_img=action_dist_img,
+                         ac=alg_step.info))
+
+Basically, what we'd like to do is taking the predicted action and action distribution
+from the ``AlgStep`` of ``ActorCriticAlgorithm``, and call :func:`.render.render_action`
+and :func:`.render.render_action_distribution` to obtain two :class:`~.render.Image`
+objects. The final step is to make sure to put the objects in the ``info`` field of
+the returned ``AlgStep``. It doesn't matter how we organize the two objects in ``info``:
+as long as they are in it, :mod:`~.alf.bin.play` will find and display them.
+
+Note that we created a namescope of "ACRender" when calling the rendering functions.
+This namescope usage is exactly the same with the namescope for summary functions:
+it will prefix all rendered image names with "ACRender/". These image names will
+be displayed as labels in the final video.
+
+Finally, we tell ALF to use our newly defined algorithm:
+
+.. code-block:: python
+
+    alf.config(
+        'TrainerConfig',
+        algorithm_ctor=partial(
+            ACRenderAlgorithm, optimizer=alf.optimizers.Adam(lr=1e-3)))
+
+Now let's train and play this conf file:
+
+.. code-block:: bash
+
+    python -m alf.bin.train --root_dir /tmp/ac_render --conf <ALF_ROOT>/alf/examples/tutorial/ac_render_conf.py
```

Comment:
Maybe add a comment between those two lines to give our readers some expectation of how long the first command runs?

```bash
python -m alf.bin.train --root_dir /tmp/ac_render --conf <ALF_ROOT>/alf/examples/tutorial/ac_render_conf.py
# After about 1 minute the above training will finish. Run the following command to play the trained model afterwards.
python -m alf.bin.play --root_dir /tmp/ac_render --num_episodes 1 --record_file /tmp/tmp.mp4 --alg_render
```